### PR TITLE
Convert special tests to testsets

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: test/linalg/special.jl [hunk 1]
```diff
@@ -1,59 +1,54 @@
...
 
-debug && println("Test interconversion between special matrix types")
-let a=[1.0:n;]
-   A=Diagonal(a)
-   for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, Matrix]
-       debug && println("newtype is $(newtype)")
+@testset "Interconversion between special matrix types" begin
+   a = [1.0:n;]
+   A = Diagonal(a)
+   @testset for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, Matrix]
```
Comment (fredrikekre) [inline test/linalg/special.jl]: I feel like we can live without the inner testsets in the `"Interconversion between special matrix types"` testset. Here for instance, we make 5 testsets with 1 test inside. There is also quite an overhead for each testset, so would be nice to limit the number of sets. As I said above, I think we can remove the inner testsets here, or if you like it better, simplify to something like:

```julia
@testset "converstion from Diagonal" begin
    for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, Matrix]
        @test full(convert(newtype, A)) == full(A)
    end
end
```
such that at least all of these tests gather in the same set.

File: test/linalg/special.jl [hunk 2]
```diff
@@ -1,59 +1,54 @@
...
    end
 
-   for isupper in (true, false)
-       debug && println("isupper is $(isupper)")
-       A=Bidiagonal(a, [1.0:n-1;], ifelse(isupper, :U, :L))
-       for newtype in [Bidiagonal, Tridiagonal, Matrix]
-           debug && println("newtype is $(newtype)")
+   @testset for isupper in (true, false)
+       A = Bidiagonal(a, [1.0:n-1;], ifelse(isupper, :U, :L))
+       @testset for newtype in [Bidiagonal, Tridiagonal, Matrix]
```
Comment (fredrikekre) [inline test/linalg/special.jl]: same here: `@testset "conversion from Bidiagonal" begin`, or completely remove the set.

**Review by fredrikekre — COMMENTED** [review]
Same comments apply to rest of the file as well :smile:

File: test/linalg/special.jl (hunk 1 above)
Comment (tkelman) [inline test/linalg/special.jl]: looks like you got the later block in this file, but this first interconversion block is also oddly 3-spaced

**Comment by kshyatt** [comment]
Did I get everything?

**Review by fredrikekre — APPROVED** [review]
:sweden:

**Pull request merged**